common/src/main/java/com/proximyst/ban/service/impl/ImplPunishmentService.java

```diff
@@ -122,9 +122,13 @@ public ImplPunishmentService(final @NonNull IDataService dataService,
       return CompletableFuture.completedFuture(null);
     }
 
-    return this.messageService.formatMessage(
-        punishment.getPunishmentType().getApplicationMessage(punishment.getReason().isPresent()).orElse(null),
+    return this.messageService.punishmentMessage(
+        punishment
+            .getPunishmentType()
+            .getApplicationMessage(punishment.getReason().isPresent())
+            .orElseThrow(() -> new IllegalStateException("applicable punishment without punishment type")),
```

Comment:
nitpick: Exception message should probably start with a capital to be more uniform with the other exception messages.